### ISSUE TITLE
[Feature] [WebKit] support AVIF (in Windows)

### ISSUE THREAD

**Issue by tats-u** [microsoft/playwright#22781]
Safari on iOS 16+ supports AVIF but Playwright WebKit browser doesn't.
Edge also supports in Canary (possibly Dev) and the other browsers have already done in stable versions.

```
npx playwright install webkit
npx playwright wk https://tests.caniuse.com/avif
```

![image](https://user-images.githubusercontent.com/12870451/235850162-8cb8dc67-a4ef-4a29-a6a3-1a8f6f74defa.png)

**Comment by fujii** [comment]
WebKit bug: https://bugs.webkit.org/show_bug.cgi?id=261368

**Comment by donny-dont** [comment]
Would it be possible for the Playwright folks to add in a built version of `libavif` to their Windows build? We could make it so the build checks to see if the library is present and then build with support for it. I'm happy to go over how one could set this up if its doable @mxschmitt

There's a maintenance burden for us every time we add a new image codec and AVIF file support isn't needed for anything we're shipping with our WebKit builds. For example I would like to remove openjpeg from the WebKitRequirements to have one less library to deal with.

**Comment by tats-u** [comment]
@donny-dont 
> AVIF file support isn't needed for anything we're shipping with our WebKit builds.
It's now supported by all browsers except for Edge.
Lossy WebP has a problem about the color space, so new images that used to be JPEG will be AVIF after Edge adds the support for it until JPEG XL is supported by Chromium & Safari.

**Comment by donny-dont** [comment]
Myself and @fujii work on the Windows and PlayStation ports of WebKit. We have internal clients that use a build of the Windows ports but they don't use them as a general browser so AVIF support is not needed for those customers. So at this time we will not be adding `libavf` and its dependencies to the WebKitForWindows/WebKitRequirements repository.

Again this doesn't mean that the folks building out Playwright couldn't add `libavif` on top of the requirements we release. And I am pro making it so someone building out the Windows port can easily add `libavif` support so we will add the plumbing in and the Playwright folks can decide whether they want to build out `libavif`.

**Comment by tats-u** [comment]
Is libjxl in the requirement? If so why do you discriminate libavif?

**Comment by tats-u** [comment]
PR in WebKit repo: https://github.com/WebKit/WebKit/pull/17640

**Comment by donny-dont** [comment]
Hi @tats-u this was already fixed in the WebKit repo at https://github.com/WebKit/WebKit/commit/e9271eae9cb9a6fbb46598c171dc1b3d927813e3

As I've been telling you the maintainers of Playwright can just do a build of `libavif` through `vcpkg` and add that to their `WebKitLibraries/win` folder and everything should work,

**Comment by tats-u** [comment]
@donny-dont I've thought this is blocked by https://github.com/WebKit/WebKit/pull/17640, which is blocked by https://github.com/WebKitForWindows/WebKitRequirements/pull/168. I'm glad to know that the ball is in the court of Playwright now.

https://webkit.org/webkit-on-windows/

Doesn't `build-webkit`automatically fetch libavif in Windows (or Linux) via vcpkg by default, right?

**Comment by tats-u** [comment]
https://github.com/microsoft/playwright/issues/32673

Looks like Linux is opt-in, too.

**Comment by mxschmitt** [comment]
Update from the Playwright side, we are actively exploring ways to add avif support to the Windows port. We are **not** blocked by upstream WebKit project - its more that we didn't like the bundle size it was adding size wise. 

@tats-u

**Comment by fujii** [comment]
I think the upstream WebKit should adopt Playwright team's new WebKitRequirment. Do you object, @donny-dont ?

**Comment by tats-u** [comment]
If we avoid the bundle of libavif, we will have to make WebKit use WindowsAPI (I doubt it can) as the macOS and iOS ports do.
We need a list of the names of vcpkg packages recommended by WebKit other than WebKitRequirement.
It's a pitfall for browsers' developers that the AVIF support in WebKit in other than macOS is opt-in.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you have additional information not present in this issue that you think will help prioritizing it, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.

**Comment by tats-u** [comment]
AVIF has already been "_Baseline_ 2024". I have waited for _your_ additional action. Is it your answer? How terrible it is!

**Comment by tats-u** [comment]
You Microsoft employees have forgotten the fact that this WebKit in Playwright is the only way to emulate the Safari's behavior in Windows. It is really stupid that you discriminate against the OS made by **_your company_**!

**Comment by tats-u** [comment]
I started to suspect that you do not know the fact that Edge has already supported AVIF since as old as 121 even though you belong to the company that develops Edge.
